beta-src/src/components/ui/units/WDFleetIcon.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import { useTheme } from "@mui/material/styles";
+import { IconProps } from "../../../interfaces/Icons";
+import UIState from "../../../enums/UIState";
+
+const WDFleetIcon: React.FC<IconProps> = function ({
+  country,
+  iconState = UIState.NONE,
+}): React.ReactElement {
+  const theme = useTheme();
+
+  return (
+    <>
+      {iconState === UIState.NONE && (
```

Comment:
Can this just be a switch check instead of having all this conditional checks.

Reply:
I think since this is at the component level this is equivalent to a switch statement. However, this is old code and I don't think we should fix it in this PR.

Reply:
Old code but this is showing this whole file as new?

Reply:
Yes, the old WDFleetIcon.tsx was coupled with WDUnitController.tsx. I needed to decouple the SVG drawing part from the logical part so I replaced the old WDFleetIcon.tsx with this and also created WDFleet.tsx and moved them into `/components/ui/units`

Reply:
Here is the original PR: https://github.com/codazen/webDiplomacy/pull/16/files#diff-7bf87ef281b198a910047daef361a1f21d9dba39c0f696dba86c7a36769d69fa